### PR TITLE
[Doc] Add doc about config

### PR DESCRIPTION
**Pull request by LeoXing1996**

### CODE REVIEW

File: docs/en/user_guides/1_config.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+Configs consisting of components from `_base_` are called _primitive_.
+
+For all configs under the same folder, it is recommended to have only **one** _primitive_ config. All other configs should inherit from the _primitive_ config. In this way, the maximum of inheritance level is 3.
+
+For easy understanding, we recommend contributors to inherit from existing methods.
+For example, if some modification is made base on StyleGAN2, user may first inherit the basic StyleGAN2 structure by specifying `_base_ = ../styleganv2/stylegan2_c2_ffhq_256_b4x8_800k.py`, then modify the necessary fields in the config files.
+
+If you are building an entirely new method that does not share the structure with any of the existing methods, you may create a folder `xxxgan` under `configs`,
+
+Please refer to [mmcv](https://mmcv.readthedocs.io/en/latest/utils.html#config) for detailed documentation.
```
Comment (Z-Fran) [inline docs/en/user_guides/1_config.md:40]: This link is unavailable.

File: docs/en/user_guides/1_config.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+val_evaluator = dict(metrics=metrics)  # define metric in evaluation process
+test_evaluator = dict(metrics=metrics)  # define metric in test process
+```
+
+## FAQ
+
+### Ignore some fields in the base configs
+
+Sometimes, you may set `_delete_=True` to ignore some of fields in base configs.
+You may refer to [mmcv](https://mmcv.readthedocs.io/en/latest/utils.html#inherit-from-base-config-with-ignored-fields) for simple illustration.
```
Comment (Z-Fran) [inline docs/en/user_guides/1_config.md:159]: This link is unavailable.

File: docs/en/user_guides/1_config.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+You may have a careful look at [this tutorial](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/tutorials/config.md) for better understanding of this feature.
```
Comment (Z-Fran) [inline docs/en/user_guides/1_config.md:161]: This should be English docs. Refer https://github.com/open-mmlab/mmgeneration/pull/410

**Comment by LeoXing1996** [comment]
We use [url of Chinese document](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/tutorials/config.md#%E5%88%A0%E9%99%A4%E5%AD%97%E5%85%B8%E4%B8%AD%E7%9A%84-key) in this document, because we do not know the subtitle of this section in English one for it has not been added.
This will be replaced with an English one after MMEngine adds this document.

File: docs/en/user_guides/1_config.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    dict(type='PrecisionAndRecall', fake_nums=50000, prefix='PR-50K'),
+    dict(type='PerceptualPathLength', fake_nums=50000, prefix='ppl-w')
+]
+default_hooks = dict(checkpoint=dict(save_best='FID-Full-50k/fid'))  # save checkpoint has the best FID metric
+
+val_evaluator = dict(metrics=metrics)  # define metric in evaluation process
+test_evaluator = dict(metrics=metrics)  # define metric in test process
+```
+
+## FAQ
```
Comment (zengyh1900) [inline docs/en/user_guides/1_config.md:154]: move FAQ to faq.md

File: docs/en/user_guides/1_config.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+- `{schedule}`: training schedule. Including learning rate (e.g., `lr1e-3`), number of gpu and batch size is used (e.g., `8xb32`), and total iterations (e.g., `160kiter`) or number of images shown in the discriminator (e.g., `12Mimgs`).
+- `{dataset}`: dataset like `ffhq`, `lsun-car`, `celeba-hq`.
+
+## An Example of StyleGAN2
+
+To help the users have a basic idea of a complete config and the modules in a modern detection system,
+we make brief comments on the config of Stylegan2 at 256x256 scale.
+For more detailed usage and the corresponding alternative for each module, please refer to the API documentation and the [tutorial in MMDetection](https://github.com/open-mmlab/mmdetection/blob/master/docs/en/tutorials/config.md).
+
+```python
```
Comment (zengyh1900) [inline docs/en/user_guides/1_config.md:64]: you can break this long snippet of config into several parts and explain each part briefly, e.g., mode, train_dataloader, test_dataloader, etc.

File: docs/en/user_guides/1_config.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+
+- Update keys inside a list of configs.
+
+  Some config dicts are composed as a list in your config. For example, the training pipeline `data.train.pipeline` is normally a list
+  e.g. `[dict(type='LoadImageFromFile'), ...]`. If you want to change `'LoadImageFromFile'` to `'LoadImageFromWebcam'` in the pipeline,
+  you may specify `--cfg-options data.train.pipeline.0.type=LoadImageFromWebcam`.
+
+- Update values of list/tuples.
+
+  If the value to be updated is a list or a tuple. For example, the config file normally sets `workflow=[('train', 1)]`. If you want to
```
Comment (Z-Fran) [inline docs/en/user_guides/1_config.md:23]: old config system.

File: docs/en/user_guides/1_config.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    by_epoch=False,  # The scheduler is called by iteration
+    start_factor=0.0002,  # The number we multiply parameter value in the first iteration
+    end_factor=0,  # The number we multiply parameter value at the end of linear changing process.
+    begin=40000,  # The start iteration of the scheduler
+    end=80000)  # The end iteration of the scheduler
+```
+
+### Hook config
+
+Users can attach hooks to training, validation, and testing loops to insert some oprations during running. There are two different hook fields, one is `default_hooks` and the other is `custom_hooks`.
```
Comment (Z-Fran) [inline docs/en/user_guides/1_config.md:209]: typo `oprations`

File: docs/en/user_guides/1_config.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+        type='CheckpointHook',
+        interval=10000,
+        by_epoch=False,
+        less_keys=['FID-Full-50k/fid'],
+        greater_keys=['IS-50k/is'],
+        save_optimizer=True,
+        save_best='FID-Full-50k/fid'))
+```
+
+`default_hooks` is a list of hook configs. Users can develop there own hooks and insert them in this field.
```
Comment (Z-Fran) [inline docs/en/user_guides/1_config.md:227]: may be custom_hook

**Pull request merged**